pkg/activator/net/revision_backends.go

```diff
@@ -215,7 +215,7 @@ func (rw *revisionWatcher) probePodIPs(dests sets.String) (sets.String, bool, er
 	ctx, cancel := context.WithTimeout(context.Background(), probeTimeout)
 	defer cancel()
 
-	var probeGroup errgroup.Group
+	probeGroup, ctx := errgroup.WithContext(ctx)
```

Comment:
This behavior of the replacement `ctx` isn't obvious to me so I am assuming it's not obvious for many. Can you please add a comment.

Thanks for this great change.

Reply:
If you agree, add a short comment to other instances as well.

Reply:
Would it help if we explicitly use `egCtx` everywhere? After all, it should probably only be used inside of the goroutines anyway.

Reply:
yes, that is a good point.